src/js/angular/similarity/controllers/create-index.controller.js

```diff
@@ -12,741 +24,687 @@ CreateSimilarityIdxCtrl.$inject = ['$scope', 'toastr', '$uibModal', '$timeout',
-        }
-    };
 
+        if (similarityIndexInfo.isPredicationType()) {
+            return SimilarityRestService.getIndexes()
+                .then((response) => {
+                    const indexesNotInProgress = mapIndexesResponseToSimilarityIndex(response.data)
+                        .filter((similarityIndex) => similarityIndex.isTextLiteralType() && (similarityIndex.isBuiltStatus() || similarityIndex.isOutdatedStatus()))
+                        .map((similarityIndex) => similarityIndex.name);
 
-    $scope.getResultsDescription = function () {
-    };
+                    $scope.literalIndexes = ['no-index'].concat(indexesNotInProgress);
 
-    $scope.getUpdateDescription = function () {
-    };
+                    if (similarityIndexInfo.getInputIndex() === undefined) {
+                        const desiredIdx = getAndRemoveOption(similarityIndexInfo.getSimilarityIndex(), '-input_index');
+                        if (desiredIdx !== undefined) {
+                            for (let j = 0; j < $scope.literalIndexes.length; j++) {
+                                if (desiredIdx === $scope.literalIndexes[j]) {
+                                    similarityIndexInfo.setInputIndex($scope.literalIndexes[j]);
+                                }
+                            }
+                        }
+                    }
+                    if (similarityIndexInfo.getInputIndex() === undefined) {
+                        similarityIndexInfo.setInputIndex($scope.literalIndexes[0]);
+                    }
+                    return similarityIndexInfo;
+                })
+                .catch(function (error) {
+                    const msg = getError(error);
+                    toastr.error(msg, $translate.instant('similarity.could.not.get.indexes.error'));
+                });
+        }
 
-    $scope.getStaleWarningMessage = function () {
-    };
+        return Promise.resolve(similarityIndexInfo);
+    }
 
-    $scope.saveSearchQuery = function () {
-        // Should validate that query is SELECT
-        if (window.editor.getQueryType() !== 'SELECT') {
-            toastr.error($translate.instant('similarity.index.select.queries.constraint'));
-            return;
+    const updateYasguiComponent = (config) => {
+        const defaultConfig = {
+            endpoint: getQueryEndpoint(),
+            componentId: 'create-index',
+            showEditorTabs: false,
+            showToolbar: false,
+            showResultTabs: false,
+            showYasqeActionButtons: false,
+            showQueryButton: false,
+            prefixes: $scope.usedPrefixes,
+            render: RenderingMode.YASQE,
+            yasqeActionButtons: $scope.isEditViewMode() || !$scope.similarityIndexInfo.isDataQueryTypeSelected() ? DISABLE_YASQE_BUTTONS_CONFIGURATION : INFERRED_AND_SAME_AS_BUTTONS_CONFIGURATION,
+            maxPersistentResponseSize: 0,
+            yasqeMode: $scope.canEditActiveRepo ? YasqeMode.WRITE : YasqeMode.PROTECTED,
```

Comment:
Are you sure this is properly initialized? You should be able to edit the query in the editor if repo is writable.

Reply:
I wrote a TODO comment about this. I have to add a 'Change repository' event listener and then recreate the component according to the new repository.